src/goose.rs

```diff
@@ -317,49 +316,66 @@ macro_rules! taskset {
     };
 }
 
+/// Goose tasks return a result, which is empty on success, or contains a GooseTaskError
+/// on error.
+pub type GooseTaskResult = Result<(), GooseTaskError>;
+
 /// Definition of all errors Goose Tasks can return.
 #[derive(Debug)]
 pub enum GooseTaskError {
+    /// Contains a reqwest::Error.
     Reqwest(reqwest::Error),
+    /// Contains a url::ParseError.
     Url(url::ParseError),
-    /// The request failed.
-    RequestFailed,
+    /// The request failed. The `GooseRawRequest` that failed can be found in
+    /// `.raw_request`.
+    RequestFailed { raw_request: GooseRawRequest },
     /// The request was canceled (this happens when the throttle is enabled and
-    /// the load test finished).
-    RequestCanceled,
-    /// The request succeeded, but there was an error recording the statistics.
-    StatsFailed,
+    /// the load test finished). The mpsc SendError can be found in `.source`.
+    /// A `GooseRawRequest` has not yet been constructed, so is not available in
+    /// this error.
+    RequestCanceled {
+        source: mpsc::error::SendError<bool>,
+    },
+    /// There was an error sending the statistics for a request to the parent thread.
+    /// The `GooseRawRequest` that was not recorded can be extracted from the error
+    /// chain, available inside `.source`.
+    StatsFailed {
+        source: mpsc::error::SendError<GooseRawRequest>,
+    },
     /// Attempt to send debug detail to logger failed.
-    LoggerFailed,
-    /// Attempted an unrecognized HTTP request method.
-    InvalidMethod,
+    /// There was an error sending debug information to the logger thread. The
+    /// `GooseDebug` that was not logged can be extracted from the error chain,
+    /// available inside `.source`.
+    LoggerFailed {
+        source: mpsc::error::SendError<Option<GooseDebug>>,
+    },
+    /// Attempted an unrecognized HTTP request method. The unrecognized method
+    /// is available in `.method`.
+    InvalidMethod { method: Method },
 }
 
-/// Goose tasks return a result, which is empty on success, or contains a GooseTaskError
-/// on error.
-pub type GooseTaskResult = Result<(), GooseTaskError>;
-
+// Define how to display errors.
 impl fmt::Display for GooseTaskError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(&self, f)
```

Comment:
@jeremyandrews This will create an infinite loop as per https://github.com/rust-lang/rust/issues/74892.

Reply:
It only works right now as we use the Debug and not the Display output for errors when returning from main.

Reply:
see #128